### DISCUSSION TITLE
Add support for "json" secrets

### DISCUSSION THREAD

**Discussion by EnergyNick** [mcrio/Configuration.Provider.Docker.Secrets#6] [Ideas]
Hi!
I would like to suggest adding support for secrets that provide a json file as a value. 
It would be more convenient to work in some cases 🤔

**Comment by mcrio** [comment]
Hi, how do you generate the json file, is it something that comes with Docker Swarm / Kubernetes?

**Reply by EnergyNick** [reply]
No)
I wanted to suggest parsing secrets (with json inside value) like FromJson, but have not yet come up with the correct API for this

**Reply by mcrio** [reply]
I see, you want to keep multiple secrets as a json string value in one file. 

I'm not sure this approach is common or encouraged in the official guidelines (let me know if I'm wrong). Usually there aren't that many secrets and having them separate makes it easier to track for updates. 

But anyway, have you tried using `AddJsonFile()` on that file that contains the json string. It's how the framework by default loads `appsettings.json`.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.jsonconfigurationextensions.addjsonfile?view=dotnet-plat-ext-6.0

**Reply by mcrio** [reply]
In case `AddJsonFile` does not work for some reason, what we could do is add an option for providing an optional custom handler that determines how the secret is added to the configuration. That way you could tell how to process the json and what values to add. This approach would be most flexible.
